### PR TITLE
Flakes in `Security2180Test`

### PR DESCRIPTION
**Pull request by jglick**
From a PCT run:

```
java.io.IOException: …/matrix-auth-plugin/target/tmp/j h5449561387972401643/jobs/folder0/jobs/item1/builds/permalinks
	at org.jvnet.hudson.test.TemporaryDirectoryAllocator.delete(TemporaryDirectoryAllocator.java:144)
	at org.jvnet.hudson.test.TemporaryDirectoryAllocator.delete(TemporaryDirectoryAllocator.java:131)
	at org.jvnet.hudson.test.TemporaryDirectoryAllocator.delete(TemporaryDirectoryAllocator.java:131)
	at org.jvnet.hudson.test.TemporaryDirectoryAllocator.delete(TemporaryDirectoryAllocator.java:131)
	at org.jvnet.hudson.test.TemporaryDirectoryAllocator.delete(TemporaryDirectoryAllocator.java:131)
	at org.jvnet.hudson.test.TemporaryDirectoryAllocator.delete(TemporaryDirectoryAllocator.java:131)
	at org.jvnet.hudson.test.TemporaryDirectoryAllocator.dispose(TemporaryDirectoryAllocator.java:99)
	at org.jvnet.hudson.test.TestEnvironment.dispose(TestEnvironment.java:84)
	at org.jvnet.hudson.test.JenkinsRule.after(JenkinsRule.java:525)
	at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:626)
	at …
Caused by: java.nio.file.DirectoryNotEmptyException: …/matrix-auth-plugin/target/tmp/j h5449561387972401643/jobs/folder0/jobs/item1/builds
	at java.base/sun.nio.fs.UnixFileSystemProvider.implDelete(UnixFileSystemProvider.java:247)
	at java.base/sun.nio.fs.AbstractFileSystemProvider.deleteIfExists(AbstractFileSystemProvider.java:110)
	at java.base/java.nio.file.Files.deleteIfExists(Files.java:1181)
	at org.jvnet.hudson.test.TemporaryDirectoryAllocator.delete(TemporaryDirectoryAllocator.java:141)
	... 13 more
```

https://github.com/jenkinsci/jenkins-test-harness/pull/198 is highlighting anomalous behavior here where builds are running after the ends of tests. Other puzzling flakes have been reported in this class that might be caused by builds from one Jenkins session completing while another is running.

Most expedient in this case to just look for all running builds since from a quick glance there were three or four different places where builds were launched, and from differently-placed projects.

### CODE REVIEW

**Review by jtnord — APPROVED** [review]

**Comment by jglick** [comment]
@daniel-beck?

**Comment by daniel-beck** [comment]
@jglick Thanks for the PR. Could you clarify what you need? Acknowledging the PR? Merge into `master`? New plugin release?

**Comment by jglick** [comment]
New plugin release please.

**Comment by jglick** [comment]
@daniel-beck ?

File: src/test/java/org/jenkinsci/plugins/matrixauth/Security2180Test.java [hunk 1]
```diff
@@ -43,9 +47,26 @@ public class Security2180Test {
     private static final Map<Permission, Set<String>> ANONYMOUS_CAN_ITEM_READ =
             Collections.singletonMap(Item.READ, Collections.singleton("anonymous"));
 
+    private static final Logger LOGGER = Logger.getLogger(Security2180Test.class.getName());
+
+    @ClassRule
+    public static BuildWatcher watcher = new BuildWatcher();
```
Comment (daniel-beck) [inline src/test/java/org/jenkinsci/plugins/matrixauth/Security2180Test.java:53]: What is this for? https://github.com/jenkinsci/jenkins-test-harness/blob/a7b41a28620768c3f3b1a6bbcb909108e03cbb21/src/main/java/org/jvnet/hudson/test/BuildWatcher.java#L43 makes it look more like a test development utility.

**Review by daniel-beck — APPROVED** [review]

File: src/test/java/org/jenkinsci/plugins/matrixauth/Security2180Test.java (hunk 1 above)
Comment (jglick) [reply src/test/java/org/jenkinsci/plugins/matrixauth/Security2180Test.java:53]: Well yes, but you might some day wish to edit the test.

**Pull request merged**